### PR TITLE
Add form for User Profile attribute groups

### PR DESCRIPTION
**Pull request by jonkoops**
Implements the Attribute Groups under the User Profile as per [design](https://marvelapp.com/prototype/5eed532/section/1460646). This allows Attribute Groups to be created and edited using a form.

## Verification Steps
1. Enable 'User Profile' under the realm settings and save the settings.
2. Go to 'User Profile' :arrow_right: 'Attributes Groups' and create a group.
3. Verify that you return to the overview and the new entry is displayed.
4. Edit the newly created entry and verify that it is saved correctly.

Tests will follow in a separate PR (see #2022).

### CODE REVIEW

**Comment by jonkoops** [comment]
@yih-wang @xianli123 could y'all verify that this feature is working as expected from a UDX point of view?

**Comment by yih-wang** [comment]
Hi @jonkoops , here are some issues I found in the current implementation:

Create group
1. Breadcrumb is missing ([design](https://marvelapp.com/prototype/5eed532/screen/82246803))
2. Tooltips are missing for the Name, Display name and Display description (all the same with the old console)
3. "Add annotation" is disabled until the current row (key) is filled 

Groups list

4. Group names in the list should be clickable (clicking the group name can edit the group)

Edit group ([design](https://marvelapp.com/prototype/5eed532/screen/85071685) is missing before)

5. Group name cannot be edited once it's created
6. Header is wrong
7. Same issues with the Create page (breadcrumb, tooltips, "Add annotation" behavior)

**Review by yih-wang — CHANGES_REQUESTED** [review]
Hi @jonkoops , here are some issues I found in the current implementation:

Create group

1. Breadcrumb is missing ([design](https://marvelapp.com/prototype/5eed532/screen/82246803))
2. Tooltips are missing for the Name, Display name and Display description (all the same with the old console)
3. "Add annotation" should be disabled until the current row (key) is filled

Groups list

4. Group names in the list should be clickable (clicking the group name can edit the group)

Edit group ([design](https://marvelapp.com/prototype/5eed532/screen/85071685) is missing before)

5. Group name cannot be edited once it's created
6. Header is wrong
7. Same issues with the Create page (breadcrumb, tooltips, "Add annotation" behavior)

**Comment by jonkoops** [comment]
@yih-wang Thanks for the feedback, I will try to take a look at it today.

File: src/realm-settings/routes/EditAttributesGroup.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import { generatePath } from "react-router-dom";
+import type { RouteDef } from "../../route-config";
+
+export type EditAttributesGroupParams = {
+  realm: string;
+  name: string;
+};
+
+export const EditAttributesGroupRoute: RouteDef = {
+  path: "/:realm/realm-settings/userProfile/attributesGroup/edit/:name",
```
Comment (edewit) [inline src/realm-settings/routes/EditAttributesGroup.ts:12]: I think it's nicer to only use lower case in urls, WDYT?

File: src/realm-settings/user-profile/AttributesGroupForm.tsx [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+  name: "",
+};
+
+export default function AttributesGroupForm() {
+  const { t } = useTranslation();
+  const { realm } = useRealm();
+  const { config, save } = useUserProfile();
+  const history = useHistory();
+  const params = useParams<Partial<EditAttributesGroupParams>>();
+  const form = useForm<FormFields>({ defaultValues });
```
Comment (edewit) [inline src/realm-settings/user-profile/AttributesGroupForm.tsx:82]: Why not use a destructure here:
```suggestion
  const { control, register, errors, reset } = useForm<FormFields>({ defaultValues });
```

File: src/realm-settings/user-profile/AttributesGroupForm.tsx [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+            />
+          </FormGroup>
+          <TextContent>
+            <Text component="h2">{t("attributes-group:annotationsText")}</Text>
+          </TextContent>
+          <FormGroup
+            label={t("attributes-group:annotationsText")}
+            fieldId="kc-annotations"
+          >
+            <Flex direction={{ default: "column" }}>
```
Comment (edewit) [inline src/realm-settings/user-profile/AttributesGroupForm.tsx:194]: There is a pre made `<AttributeInput` component that does this

**Review by edewit — CHANGES_REQUESTED** [review]

File: src/realm-settings/routes/EditAttributesGroup.ts (hunk 1 above)
Comment (jonkoops) [reply src/realm-settings/routes/EditAttributesGroup.ts:12]: I agree, this is just very inconsistent everywhere already. Perhaps we can log it as an issue to resolve this for all routes in one go?

File: src/realm-settings/user-profile/AttributesGroupForm.tsx (hunk 3 above)
Comment (jonkoops) [reply src/realm-settings/user-profile/AttributesGroupForm.tsx:194]: I did mess around with this but ended up going for this solution. Let me check if I can implement it.

File: src/realm-settings/user-profile/AttributesGroupForm.tsx (hunk 2 above)
Comment (edewit) [reply src/realm-settings/user-profile/AttributesGroupForm.tsx:82]: okay, I can see why you feel this way, atm there are 50 components that already do it like that

File: src/realm-settings/user-profile/AttributesGroupForm.tsx (hunk 3 above)
Comment (edewit) [reply src/realm-settings/user-profile/AttributesGroupForm.tsx:194]: it should be a drop in replacement, although it got slightly polluted with the new `<Dropdown ` keys I think we might want to get rid of those.

**Review by agagancarczyk — APPROVED** [review]

**Review by edewit — APPROVED** [review]

File: src/realm-settings/user-profile/AttributesGroupForm.tsx (hunk 2 above)
Comment (jonkoops) [reply src/realm-settings/user-profile/AttributesGroupForm.tsx:82]: Yeah, that's unfortunate.

**Comment by jonkoops** [comment]
@yih-wang I've resolved all the issues you mentioned, would you mind taking another look?

File: src/realm-settings/user-profile/AttributesGroupForm.tsx (hunk 3 above)
Comment (jonkoops) [reply src/realm-settings/user-profile/AttributesGroupForm.tsx:242]: I'll do some work to clean this up after the PR has been merged. I'd like to make the component a bit more re-usable.

**Review by edewit — APPROVED** [review]

**Review by agagancarczyk — APPROVED** [review]

**Review by mfrances17 — APPROVED** [review]
LGTM, tested fine - good job

**Comment by yih-wang** [comment]
@jonkoops The updates look good! I only found a low-impact issue that won't block the merge. Raised a separate issue #2167 to track that.

**Comment by jonkoops** [comment]
@yih-wang Great, thanks for testing this!

**Pull request merged**